docker-compose.yml

```diff
@@ -77,5 +75,80 @@ services:
     extra_hosts:
       - "host.docker.internal:host-gateway"
 
+  monitoring_otel_collector:
+    container_name: monitoring_otel_collector
+    image: otel/opentelemetry-collector-contrib:0.98.0
+    networks:
+      - altinntestlocal_network
+    command:
+      - --config=/etc/otel-collector.yaml
+    volumes:
+      - ./infra/otel-collector.yaml:/etc/otel-collector.yaml
+    ports:
+      - "1888:1888"   # pprof extension
+      - "8888:8888"   # Prometheus metrics exposed by the collector
+      - "8889:8889"   # Prometheus exporter metrics
+      - "13133:13133" # health_check extension
+      - "4317:4317"   # OTLP gRPC receiver
+      - "55679:55679" # zpages extension
+    profiles: ["monitoring"]
+
+  monitoring_tempo:
+    container_name: monitoring_tempo
+    image: grafana/tempo:2.4.1
+    networks:
+      - altinntestlocal_network
+    command:
+      - -config.file=/etc/tempo.yaml
+      - -log.level=error
+    volumes:
+      - ./infra/tempo.yaml:/etc/tempo.yaml
+    profiles: ["monitoring"]
+
+  monitoring_mimir:
+    container_name: monitoring_mimir
+    image: grafana/mimir:2.12.0
+    networks:
+      - altinntestlocal_network
+    command:
+      - -config.file=/etc/mimir.yaml
+      - -target=all
+      - -log.level=error
+    volumes:
+      - ./infra/mimir.yaml:/etc/mimir.yaml
+    profiles: ["monitoring"]
+
+  monitoring_loki:
+    container_name: monitoring_loki
+    image: grafana/loki:3.0.0
+    networks:
+      - altinntestlocal_network
+    command:
+      - -config.file=/etc/loki.yaml
+      - -target=all
+      - -log.level=error
+    volumes:
+      - ./infra/loki.yaml:/etc/loki.yaml
+    profiles: ["monitoring"]
+
+  monitoring_grafana:
+    container_name: monitoring_grafana
+    image: grafana/grafana:10.4.2
+    networks:
+      - altinntestlocal_network
+    volumes:
+      - ./infra/grafana-datasources.yaml:/etc/grafana/provisioning/datasources/datasources.yaml
+      - ./infra/grafana-dashboards.yaml:/etc/grafana/provisioning/dashboards/dashboards.yaml
+      - ./infra/grafana-dashboards:/var/lib/grafana/dashboards
+    environment:
+      - GF_AUTH_ANONYMOUS_ENABLED=true
+      - GF_AUTH_ANONYMOUS_ORG_ROLE=Admin
+      - GF_AUTH_DISABLE_LOGIN_FORM=true
+      - GF_FEATURE_TOGGLES_ENABLE="traceqlEditor traceToMetrics metricsSummary"
+      - GF_LOG_LEVEL=error
+    ports:
+      - "3000:3000"
```

Comment:
Maybe consider a more random looking port number in case someone runs anything else on 3000?

Reply:
Or run this through the nginx "load balancer" (proxy), so that it does not need to be on a separate port.

Reply:
Went with the proxy approach 👍